app/frontend/components/pages/sendAda/sendAdaPage.tsx

```diff
@@ -40,208 +46,364 @@ interface Props {
+        {type === AssetFamily.TOKEN ? quantity : printAda(Math.abs(quantity) as Lovelace)}
+      </div>
+    </div>
+    {policyId && (
+      <div className="multi-asset-hash">
+        <span className="ellipsis">{policyId.slice(0, -6)}</span>
+        <span>{policyId.slice(-6)}</span>
+      </div>
+    )}
+  </div>
+)
+
+const SendAdaPage = ({
+  sendResponse,
+  sendAddress,
+  sendAddressValidationError,
+  sendAmount,
+  sendAmountValidationError,
+  updateAddress,
+  updateAmount,
+  confirmTransaction,
+  shouldShowConfirmTransactionDialog,
+  feeRecalculating,
+  sendMaxFunds,
+  conversionRates,
+  sendTransactionSummary: summary,
+  transactionFee,
+  txSuccessTab,
+  balance,
+  isModal, // TODO: remove
+  title,
+  sourceAccountIndex,
+  targetAccountIndex,
+  setSourceAccount,
+  setTargetAccount,
+  switchSourceAndTargetAccounts,
+  tokenBalance,
+}: Props) => {
+  const amountField = useRef<HTMLInputElement>(null)
+  const submitTxBtn = useRef<HTMLButtonElement>(null)
+  const sendCardDiv = useRef<HTMLDivElement>(null)
+
+  const sendFormValidationError = sendAddressValidationError || sendAmountValidationError
 
-  render({
-    sendResponse,
-    sendAddress,
-    sendAddressValidationError,
-    sendAmount,
-    sendAmountValidationError,
-    donationAmountValidationError,
-    updateAddress,
-    updateAmount,
-    confirmTransaction,
-    shouldShowConfirmTransactionDialog,
-    feeRecalculating,
-    sendMaxFunds,
-    shouldShowThanksForDonation,
-    closeThanksForDonationModal,
-    coinsAmount,
-    shouldShowCustomDonationInput,
-    maxDonationAmount,
-    conversionRates,
-    sendTransactionSummary: summary,
-    transactionFee,
-    txSuccessTab,
-    balance,
-    showDonationFields,
-    isModal, // TODO: remove
-    title,
-    sourceAccountIndex,
-    targetAccountIndex,
-    setSourceAccount,
-    setTargetAccount,
-    switchSourceAndTargetAccounts,
-  }) {
-    const sendFormValidationError =
-      sendAddressValidationError || sendAmountValidationError || donationAmountValidationError
+  const enableSubmit = sendAmount.fieldValue && sendAddress && !sendFormValidationError
+  const isSendAddressValid = !sendAddressValidationError && sendAddress !== ''
 
-    const enableSubmit = sendAmount && sendAddress && !sendFormValidationError
-    const isDonationSufficient = maxDonationAmount >= toCoins(ADALITE_MIN_DONATION_VALUE)
-    const isSendAddressValid = !sendAddressValidationError && sendAddress !== ''
-    const total = summary.amount + transactionFee + summary.donation
+  const totalLovelace =
+    ((summary.coins + summary.fee + summary.minimalLovelaceAmount) as Lovelace) || (0 as Lovelace)
+  const totalTokens = summary.token
+
+  const adaAsset: DropdownAssetItem = {
+    type: AssetFamily.ADA,
+    policyId: null,
+    assetName: 'ADA',
+    assetNameHex: null,
+    quantity: balance,
+    star: true,
+  }
+  const dropdownAssetItems: Array<DropdownAssetItem> = useMemo(
```

Comment:
will the data cached by `useMemo()` be updated after the user sends/receives tokens?

Reply:
I tested it manually by sending a new token and appears that yes :thinking:

Reply:
ahhh, I see, it's a preact hooks native helper and `[adaAsset, tokenBalance]` below specifies that the cache is invalidated any time any of those entries changes: https://preactjs.com/guide/v10/hooks/#usememo